### PR TITLE
[Serializer] Replace BadMethodCallException by MappingException

### PR DESCRIPTION
**Pull request by dunglas**
| Q | A |
| --- | --- |
| Bug fix? | no |
| New feature? | no |
| BC breaks? | no (`AnnotationLoader` will be introduced in 2.7) |
| Deprecations? | no |
| Tests pass? | yes |
| Fixed tickets | n/a |
| License | MIT |
| Doc PR | n/a |

### CODE REVIEW

File: src/Symfony/Component/Serializer/Mapping/Loader/AnnotationLoader.php [hunk 1]
```diff
@@ -67,7 +68,7 @@ public function loadClassMetadata(ClassMetadata $metadata)
                                 $metadata->addAttributeGroup(lcfirst($matches[2]), $group);
                             }
                         } else {
-                            throw new \BadMethodCallException(sprintf('Groups on "%s::%s" cannot be added. Groups can only be added on methods beginning with "get" or "is".', $className, $method->name));
+                            throw new MappingException(sprintf('Groups on "%s::%s" cannot be added. Groups can only be added on methods beginning with "get" or "is".', $className, $method->name));
```
Comment (Tobion) [inline src/Symfony/Component/Serializer/Mapping/Loader/AnnotationLoader.php]: This looks more like a LogicException than a RuntimeException (which MappingException extends from). But MappingException is indeed more consistent with the other loaders. But LoaderInterface doesn't mention what exception can be thrown.

Comment (dunglas) [inline src/Symfony/Component/Serializer/Mapping/Loader/AnnotationLoader.php]: I was thinking about the same thing. Another argument in favor of using a `MappingException`: it's what the Validator Annotation Loader throws https://github.com/symfony/symfony/blob/2.7/src/Symfony/Component/Validator/Mapping/Loader/AnnotationLoader.php#L84

**Comment by dunglas** [comment]
The Travis error is not related with this PR.

**Comment by dunglas** [comment]
ping @symfony/deciders

**Comment by Tobion** [comment]
:+1:

**Pull request merged**